Striker/Striker/Striker_logic.ino

```diff
@@ -0,0 +1,141 @@
+    Serial1.begin(9600);
+}
+
+void loop()
+{
+    bno_angle = serialComm.Receive(RECEIVE_BNO);
+    ball_angle = serialComm.Receive(RECEIVE_BALL_ANGLE);
+    ball_distance = serialComm.Receive(RECEIVE_BALL_DISTANCE);
+    goal_angle = serialComm.Receive(RECEIVE_GOAL_ANGLE);
+    goal_distance = serialComm.Receive(RECEIVE_GOAL_DISTANCE);
+
+    double speed_w = pid_w.Calculate(target_angle, bno_angle);
+    double speed_t_goal = pid_t_goal.Calculate(180, goal_angle);
+    double speed_t_ball = pid_t_ball.Calculate(0, ball_distance);
+
+//--------------------------------------- Separate coordinate plane ---------------------------------//
+
+    if (ball_angle < 180)
+    {
+        ball_angle_180 = -ball_angle;
+    }
+    else if (ball_angle > 180)
+    {
+        ball_angle_180 = 360 - ball_angle;
+    }
+
+//--------------------------------------- Ball found logic ---------------------------------------//
+
+    if (ball_angle == 0)
+    {
+        ball_found = false;
+    }
+    else
+    {
+        ball_found = true;
+        last_time_ball_seen = millis();
+        Serial.println("Ball found");
+    }
+
+//--------------------------------------- Goal found logic ---------------------------------------//
+
+    if(goal_angle == 0)
+    {
+        goal_found = false;
+    }
+    else
+    {
+        goal_found = true;
+        //Serial.println("GOAL found");
+    }
+
+//--------------------------------------- Has ball logic ----------------------------------------//
+
+    if (ball_found && ball_distance < 20 && millis() - last_time_ball_seen < time_threshold && ball_angle_180 > -20 && ball_angle_180 < 20)
+    {
+        has_ball = true;
+        Serial.println("HAS BAALLL");
+    }
+    else
+    {
+        has_ball = false;
+    }
+
+//--------------------------------------- Scoring goal logic -------------------------------------//
+
+    if(has_ball && goal_found && goal_distance > 30)
+    {
+        // move towards goal
+        myMotors.MoveMotorsImu(0, abs(speed_t_ball), speed_w);
+    }//peruana
+    else if(has_ball && goal_found && goal_distance < 30)
+    {
+        // shoot
+        myMotors.MoveMotorsImu(360 - goal_angle, abs(speed_t_ball), speed_w);
+    }
+    else if (has_ball && !goal_found)
+    {
+        // logica de no tener porteria
+    }
+
+//--------------------------------------- Follow ball logic --------------------------------------------//
+
+    if (ball_found)
+    {
+        if (ball_angle_180 > -15 && ball_angle_180 < 15)
+        {
+            myMotors.MoveMotorsImu(0, abs(speed_t_ball), speed_w);
+        }
+        else
+        {
+            ball_angle = 360 - ball_angle;
+            double differential = ball_angle * 0.15;
+            ponderated_angle = ball_angle - differential;
+            ponderated_angle = ball_angle > 180 ? ball_angle - differential : ball_angle + differential;
+            myMotors.MoveMotorsImu(ponderated_angle, abs(speed_t_ball), speed_w);
+        }
+    }
```

Comment:
If after more tuning of the movements the robot seems not optimal, maybe you could make the angle also dependant on the `ball_distance`

Reply:
We have to make more tests today, still we think the constants are the ones which need to be tuned. As for the ball distance, we do consider it in the `has_ball` logic.